### PR TITLE
TSPS-231 update exceptions thrown in flight to be deserializable

### PR DESCRIPTION
**Pull request by jsotobroad**
### Description 

These 3 exceptions are now deserializable when they are thrown in a flight (you can see in the test I changed that the exception we were expecting to fail now succeeds so I changed it to an exception that isnt thrown at all)

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-231

### CODE REVIEW

File: service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckCbasHealthStep.java [hunk 1]
```diff
@@ -36,8 +35,7 @@ public StepResult doStep(FlightContext flightContext)
         cbasService.checkHealth(cbasUri, samService.getTspsServiceAccountToken());
     if (!healthResult.isOk()) {
       return new StepResult(
-          StepStatus.STEP_RESULT_FAILURE_RETRY,
-          new CbasServiceApiException(new ApiException("CBAS is not healthy")));
```
Comment (jsotobroad) [inline service/src/main/java/bio/terra/pipelines/stairway/imputation/CheckCbasHealthStep.java:40]: it was silly that I committed this in the first place, I shouldn't need to make a whole ApiException object when all i want to pass is a string.  This feels much cleaner

File: service/src/test/java/bio/terra/pipelines/stairway/imputation/StairwayExceptionSerializerTest.java [hunk 2]
```diff
@@ -83,15 +84,16 @@ void deserialize_classNotFound() {
 
   @Test
   void deserialize_failToConstruct() {
-    // this is marked as an ApiErrorReportException, but no matching constructor can be found
+    // this is marked as an ApiErrorReportException, but no matching constructor can be found.  We
+    // dont throw this in our stairway flgiht code so it doesnt need to be deserializable
```
Comment (mmorgantaylor) [inline service/src/test/java/bio/terra/pipelines/stairway/imputation/StairwayExceptionSerializerTest.java:88]: I think I understand this after thinking about it for a while - to help future devs understand this more easily, can you add some comment here to clarify that we do throw a `LeonardoApiServiceException` but not a `LeonardoServiceException` and why those are different?

**Review by mmorgantaylor — APPROVED** [review]
looks great, just one request for comment clarification

can you also show (in the PR description) what the exception looks like now when you run a flight and hit one of these?

File: service/src/test/java/bio/terra/pipelines/stairway/imputation/StairwayExceptionSerializerTest.java (hunk 2 above)
Comment (jsotobroad) [reply service/src/test/java/bio/terra/pipelines/stairway/imputation/StairwayExceptionSerializerTest.java:88]: updated the comment!

**Comment by jsotobroad** [comment]
here is an example of what the error report for submitting to a workspace the TSPS service acccount doenst have access to looks like 
Screenshot 2024-05-15 at 11 37 52 AM

**Pull request merged**